File: LogicalThinkingPatterns/HallowRectanglePattern/HallowRectangle.py
class HallowRectangle:
    def printPattern(self, n):
        for i in range(0,n):
            pattern =""
            for j in range(0, n):
                if (i == 0 or i == n-1 or j == 0 or j == n-1):
                    pattern += "* "
                else:
                    pattern += "  "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

hallowRectangle = HallowRectangle()
hallowRectangle.run()

File: LogicalThinkingPatterns/HallowRectanglePattern/HallowRectangle.ts
class HallowReactangle {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j < n; j++) {
        if (i == 0 || i == n - 1 || j == 0 || j == n - 1) {
          pattern += "* ";
        } else {
          pattern += "  ";
        }
      }
      console.log(pattern);
    }
  }
}

const hallowRectangle = new HallowReactangle();
hallowRectangle.printPattern(5);
